pkg/sql/sqlstats/insights/ingester.go

```diff
@@ -40,6 +40,8 @@ type ConcurrentBufferIngester struct {
 	registry      *lockingRegistry
 	running       uint64
```

Comment:
Can we get rid of this field now?

Reply:
Thanks for the nudge. I avoided it earlier because I didn't take time to understand the one test that was using it, it wasn't obvious that I could remove the check. I ended up simplifying that test a lot and getting rid of the async stuff!